front/src/modules/companies/table/components/TableActionBarButtonDeleteCompanies.tsx

```diff
@@ -1,37 +1,24 @@
 import { getOperationName } from '@apollo/client/utilities';
-import { useRecoilValue } from 'recoil';
 
 import { GET_COMPANIES } from '@/companies/queries';
+import { useDeleteItems } from '@/people/hooks/deleteItems';
 import { GET_PIPELINES } from '@/pipeline/queries';
 import { IconTrash } from '@/ui/icon/index';
 import { EntityTableActionBarButton } from '@/ui/table/action-bar/components/EntityTableActionBarButton';
-import { useResetTableRowSelection } from '@/ui/table/hooks/useResetTableRowSelection';
-import { selectedRowIdsSelector } from '@/ui/table/states/selectedRowIdsSelector';
 import { useDeleteManyCompaniesMutation } from '~/generated/graphql';
 
-export function TableActionBarButtonDeleteCompanies() {
-  const selectedRowIds = useRecoilValue(selectedRowIdsSelector);
-
-  const resetRowSelection = useResetTableRowSelection();
-
+export function TableActionBarButtonDeleteCompanies({ id }: { id: string }) {
   const [deleteCompanies] = useDeleteManyCompaniesMutation({
     refetchQueries: [
       getOperationName(GET_COMPANIES) ?? '',
       getOperationName(GET_PIPELINES) ?? '',
     ],
   });
 
-  async function handleDeleteClick() {
-    const rowIdsToDelete = selectedRowIds;
-
-    resetRowSelection();
-
-    await deleteCompanies({
-      variables: {
-        ids: rowIdsToDelete,
-      },
-    });
-  }
+  const { handleDeleteClick } = useDeleteItems({
+    handleDeleteItems: deleteCompanies,
+    id,
```

Comment:
We pass in the ID so that we can make it easy for us to cancel the delete when the user moves away from the page.

Reply:
We pass in the ID from the parent component to prevent the SnackBar from being accidentally dismissed when the component re-renders